libp2p/protocols/protocol.nim

```diff
@@ -26,11 +30,16 @@ type
     codecs*: seq[string]
     handler*: LPProtoHandler ## this handler gets invoked by the protocol negotiator
     started*: bool
+    maxStreams*: Opt[int]
 
 method init*(p: LPProtocol) {.base, gcsafe.} = discard
 method start*(p: LPProtocol) {.async, base.} = p.started = true
 method stop*(p: LPProtocol) {.async, base.} = p.started = false
 
+proc hash*(protocol: LPProtocol): Hash =
```

Comment:
Is this necessary to use LPProtocol as a key in the table?

Reply:
We could also use the list of codecs, but that might do weird things when the user uses a [matcher](https://github.com/status-im/nim-libp2p/blob/b3d9360dfcc3c395cf0768567685e923587e3b77/libp2p/multistream.nim#L39)?
Or maybe we just put the CountTable directly in LPProtocol, idk, wdyt?

Reply:
Is it possible to use the object's name?